### PR TITLE
2.0 changelog

### LINKED ISSUE #292: Document breaking changes in 2.0
Let's collect some of the more subtle stuff we need to document here, before we forget about it.

- [x] The direction of managed object transformers was reversed (d063837 #153)
- [x] Property mapping needs to be explicit (#170)
- [x] JSON mapping is implicitly validated (#251)
- [x] JSON key paths are restricted to `NSDictionaries` (#275)
- [x] Minimum deployment targets are 10.9 & 8.0
- [x] Core Data support has been extracted
- [x] `<MTLModel>` protocol extracted (#219)
- [x] Error handling in value transformers (#153)
- [x] JSON transformers on `MTLJSONAdapter` (#474)
- [x] Property storage behaviors (#210)

### PR DESCRIPTION
**Pull request by jspahrsummers**
Resolves #292.

**To do:**

 - [x] Reorder entries by importance

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+### Type checking during JSON parsing
+
+`MTLJSONAdapter` now [implicitly
+validates](https://github.com/Mantle/Mantle/pull/251) the type of values
+assigned to your `<MTLModel>` objects during JSON parsing.
+
+This can be prevent errors like an `NSString` being assigned to a `BOOL`
+property.
+
+This is only a simple safety check, though, and cannot catch every kind of
```
Comment (dcaunt) [inline CHANGELOG.md]: What kinds of errors are not caught by the type checking? I wonder if we can set clearer expectations here.

Comment (jspahrsummers) [inline CHANGELOG.md]: All I mean by this is that (this) type checking helps prevent some kinds of bugs, but cannot prevent all of them.

For example, booleans and integers are both `NSNumber` types, so I don't think we would catch errors there.

**Comment by jspahrsummers** [comment]
This is ready to go, but doesn't include anything for #474 (because it may change further before it's merged).

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+direction of managed object attribute transformers has been flipped.
+
+Managed object transformers now convert _from_ managed object attributes _to_
+model properties in the forward direction. In the reverse direction, they
+convert from properties to managed object attributes.
+
+**To update:**
+
+ * Swap the forward and reverse transformation logic of any custom managed
+   object transformers.
```
Comment (robb) [inline CHANGELOG.md]: or call `mtl_reverseTransformer`?

**Comment by robb** [comment]
Just that one note :sparkling_heart:

**Comment by jspahrsummers** [comment]
:boom:

**Comment by robb** [comment]
:chart_with_upwards_trend:

File: CHANGELOG.md (hunk 1 above)
Comment (hq3595) [inline CHANGELOG.md]: +Instead, you must explicitly specify every property that should
+be mapped, and any properties omitted will not be considered for JSON
+serialization or deserialization.
I don't understand why did you make this change ?

Comment (robb) [inline CHANGELOG.md]: See #149 for more of the thought process. In a nut shell, this approach is simpler.

**Pull request merged**